ui/desktop/src/goosed.ts

```diff
@@ -25,25 +25,24 @@ export const findAvailablePort = (): Promise<number> => {
 
 // Goose process manager. Take in the app, port, and directory to start goosed in.
 // Check if goosed server is ready by polling the status endpoint
-const checkServerStatus = async (
-  port: number,
-  maxAttempts?: number,
-  interval: number = 100
-): Promise<boolean> => {
-  if (maxAttempts === undefined) {
-    const isTemporalEnabled = process.env.GOOSE_SCHEDULER_TYPE === 'temporal';
-    maxAttempts = isTemporalEnabled ? 200 : 80;
-    log.info(
-      `Using ${maxAttempts} max attempts (temporal scheduling: ${isTemporalEnabled ? 'enabled' : 'disabled'})`
-    );
-  }
+const checkServerStatus = async (port: number, secret: string): Promise<boolean> => {
+  const isTemporalEnabled = process.env.GOOSE_SCHEDULER_TYPE === 'temporal';
+  const maxAttempts = isTemporalEnabled ? 200 : 80;
+  const interval = 100;
+  log.info(
+    `Using ${maxAttempts} max attempts (temporal scheduling: ${isTemporalEnabled ? 'enabled' : 'disabled'})`
+  );
 
   const statusUrl = `http://127.0.0.1:${port}/status`;
   log.info(`Checking server status at ${statusUrl}`);
 
   for (let attempt = 1; attempt <= maxAttempts; attempt++) {
     try {
-      const response = await fetch(statusUrl);
+      const response = await fetch(statusUrl, {
+        headers: {
+          'X-Secret-Key': secret,
+        },
+      });
```

Comment:
could we use the generated API at this point or would it not be ready (or is it only on the other side?)

Reply:
looks like we don't actually add this endpoint to the generated client. It's also a bit of a mixed bag all over where we do vs don't use the generated client. I'll have a look at cleaning that up too, as this PR is kind of a logical place to do it

Reply:
yeah, it's all over the place. we should really set an alert on getSecret appearing in PRs